python/aws-lambda/security/dynamodb-filter-injection.yaml

```diff
@@ -0,0 +1,47 @@
+rules:
+- id: dynamodb-filter-injection
+  mode: taint
+  metadata:
+    cwe: "CWE-943: Improper Neutralization of Special Elements in Data Query Logic"
+    owasp: "A1: Injection"
+    category: security
+    technology:
+    - python
+    - boto3
+    - aws-lambda
+    - dynamodb
+    references:
+    - https://medium.com/appsecengineer/dynamodb-injection-1db99c2454ac
+  message: >-
+    Detected DynamoDB query filter that is tainted by `$EVENT` object. This could lead to NoSQL injection if the variable is user-controlled
+    and not properly sanitized. Explicitly assign query params instead of passing data from `$EVENT` directly to DynamoDB client.
+  pattern-sources:
+  - patterns:
+    - pattern: event
+    - pattern-inside: |
+        def $HANDLER(event, context):
+          ...
+  pattern-sanitizers:
+  - patterns:
+    - pattern: |
+        {...}
+  pattern-sinks:
+  - patterns:
+    - pattern: $SINK
+    - pattern-either:
+      - pattern-inside: $TABLE.scan(..., ScanFilter = $SINK, ...)
+      - pattern-inside: $TABLE.query(..., QueryFilter = $SINK, ...)
+    - pattern-either:
+      - patterns:
```

Comment:
this `patterns` statement is a great example of well-written `pattern-inside` usage.

Reply:
thank you!